### PR TITLE
fixes #10620

### LINKED ISSUE #10620: Structure Scan is not working.
The structure scan is not working in the latest QGroundControl App. only working in window release mode but not working in windows debug mode and also not working in android mode debug and release mode.

## Expected Behavior
click on structure scan then show the polygon for mapping.

## Current Behavior
after clicking on structure scan app hangs and crash the app and showing error

## System Information
When posting bug reports, include the following information
- Operating System: [Android, Windows]
- QGC Version: latest
- QGC build: stable
- Flight Controller: x7+
- Autopilot (with version): latest

##Screenshots
image

### PR DESCRIPTION
**Pull request by i-am-yogesh**
Added necessary checks. fixes #10620

### CODE REVIEW

File: src/MissionManager/QGCMapPolygon.cc [hunk 1]
```diff
@@ -279,7 +279,7 @@ void QGCMapPolygon::appendVertices(const QList<QGeoCoordinate>& coordinates)
         objects.append(new QGCQGeoCoordinate(coordinate, this));
         _polygonPath.append(QVariant::fromValue(coordinate));
     }
-    _polygonModel.append(objects);
+    if(!objects.empty())  _polygonModel.append(objects);
```
Comment (Davidsastresas) [inline src/MissionManager/QGCMapPolygon.cc:282]: I would place the "{}" here to follow the code style of QGC. Thanks!

File: src/MissionManager/StructureScanComplexItem.cc [hunk 2]
```diff
@@ -532,8 +532,10 @@ void StructureScanComplexItem::_rebuildFlightPolygon(void)
     int savedEntryVertex = _entryVertex;
     _entryVertex = 0;
 
-    _flightPolygon = _structurePolygon;
-    _flightPolygon.offset(_cameraCalc.distanceToSurface()->rawValue().toDouble());
+    if(_structurePolygon.isValid()){
```
Comment (Davidsastresas) [inline src/MissionManager/StructureScanComplexItem.cc:535]: Just being a bit picky, but please here we should have spaces in the "if" line:

if (_structurePolygon.isValid()) {

Thanks!

Comment (i-am-yogesh) [reply src/MissionManager/StructureScanComplexItem.cc]: Thank you for the feedback! I've addressed the code style issues in the pull request. Please let me know if there's anything else that needs improvement..

**Comment by Davidsastresas** [comment]
Thank you very much for the fixes! Can you please do a git rebase -i before merging, so we can squash the commits and the fixes don't appear as new commits? If it is too much trouble I can do that for you.

Thanks!

**Comment by i-am-yogesh** [comment]
Sorry, but can you do git rebase -i for me. Currently I will not be able to perform this operation due to some personal reasons.
Thanks a lot!

**Comment by DonLakeFlyer** [comment]
This change looks a little bit like fixing the symptom but not the actual problem. How come there are no vertices in the first place?

**Pull request closed without merging**